contracts/governance/Staking/Staking.sol

```diff
@@ -490,6 +557,28 @@ contract Staking is IStaking, WeightedStaking, ApprovalReceiver {
 		_moveDelegates(currentDelegate, delegatee, delegatorBalance, lockedTS);
 	}
 
+	// @dev delegates tokens for lock date 2 weeks later than given lock date
+	//		if message sender is a contract
+	function _delegateNext(
+		address delegator,
+		address delegatee,
+		uint256 lockedTS
+	) internal {
+		if (msg.sender.isContract()) {
+			uint256 nextLock = lockedTS.add(TWO_WEEKS);
+			address currentDelegate = delegates[delegator][nextLock];
+			if (currentDelegate != delegatee) {
+				_delegate(delegator, delegatee, nextLock);
+			}
+			// @dev workaround for the issue with a delegation of the latest stake
+			nextLock = nextLock.add(TWO_WEEKS);
+			currentDelegate = delegates[delegator][nextLock];
+			if (currentDelegate != delegatee) {
+				_delegate(delegator, delegatee, nextLock);
+			}
```

Comment:
this block duplicates lines above it. is it correct? basically, it's a loop with 2 iterations, where `2` is a magic constant - doesn't feel right.

Reply:
Added test to reproduce an issue with delegation,
it fails without this delegation.
Also added additional check before function invocation.